runtests.py

```diff
@@ -153,20 +130,85 @@ def string_to_dictionary(string):
 
 def store_variables(assignments, source, ns):
     """
-    Given variable names and xpaths, with an XML document and namespaces, find the value of the xpath in the XML and assign it to the variable name.  Modifies the global variables.
-    assignments: str A multiline mapping of varible name to xpath, as "FOO_BAR=/fizz:foo/buzz:bar[@att='val']"
+    Given variable names and xpaths, with an XML document and namespaces, find the value of the xpath in the XML and assign it to the variable name. Paths specified are writtent. Modifies the global variables. 
+    assignments: str A multiline sequence of either file paths as "/path/to/file=*" or mappings of varible names to xpaths, as "FOO_BAR=/fizz:foo/buzz:bar[@att='val']"
     source: bytes The XML against which to evaluate the xpaths
     ns: dict Key-value pairs of namespace aliases to URIs as used in the xpath
     return: None
     """
+    output_paths = []
     for line in assignments.splitlines():
         variable, xpath = line.split('=', 1)
+        if xpath == '*':
+            output_paths.append(variable)
+            continue
         xpath = variable_replace(xpath)
         value = xpath_lookup(source, xpath, ns)
         if value:
             variables[variable] = value
         else:
             variables.pop(variable, None)
+    
+    for path in output_paths:
+        with open(path, 'wb') as fw:
+            fw.write(source)
+
+    failed_store = False
+    for line in assignments.splitlines():
+        key, value = line.split('=', 1)
+        if value == '*':
+            continue
+        if not variables[key]:
+           if not failed_store:
+                print(f"#{row_number} Failed. {row['Title']}", file=sys.stderr)
+                write_to_tempfile(row_number, response.content)
+           failed_store = True
+           print(f"  #{row_number} Missing Value. Tried {value}", file=sys.stderr)
+
+def handle_tests(row_number, row, xmlfile):
+    namespaces = string_to_dictionary(row['NS'])
```

Comment:
Please add inline documentation of the purpose, inputs, outputs and side effects of this function definition.